lib/src/modules/mod.rs

```diff
@@ -210,10 +215,13 @@ pub mod mods {
     /// # Example
     /// ```rust
     /// # use yara_x;
-    /// # let data = &[];
-    /// let elf_info = yara_x::mods::invoke::<yara_x::mods::ELF>(data);
+    /// # use yara_x::ScanInputRaw;
+    /// # let data = ScanInputRaw { target: &[], meta: None };
+    /// let elf_info = yara_x::mods::invoke::<yara_x::mods::ELF>(&data);
     /// ```
-    pub fn invoke<T: protobuf::MessageFull>(data: &[u8]) -> Option<Box<T>> {
+    pub fn invoke<T: protobuf::MessageFull>(
+        data: &ScanInputRaw,
```

Comment:
Instead of changing the signature of `invoke`, add a new function:

 ```rust
pub fn invoke_with_meta(data: &[u8], meta: &[u8])
```

Besides backward-compatibility (which is not that important at this point when the API is still beta), the reason is usability. Most users of this API only need to call `invoke` without any metadata at all, an the API should facilitate the most common use case.

Reply:
implemented in [6718dd4](https://github.com/VirusTotal/yara-x/pull/183/commits/6718dd41f6da680fced617714db2d3afb1f397c2)